src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -690,6 +691,39 @@ public HttpRequestInterceptionBuilder WithInterceptionCallback(Func<HttpRequestM
         /// The current <see cref="HttpRequestInterceptionBuilder"/>.
         /// </returns>
         public HttpRequestInterceptionBuilder WithInterceptionCallback(Func<HttpRequestMessage, Task<bool>> onIntercepted)
+        {
+            _onIntercepted = DelegateHelpers.ConvertToBooleanTask(onIntercepted);
```

Comment:
As mentioned above, could maybe just adapt to the new signature instead with a discard?